libs/ledger/src/consensus/consensus.cpp

```diff
@@ -161,7 +162,16 @@ Block GetBeginningOfAeon(Block const &current, MainChain const &chain)
   // case for true genesis)
   while (!ret.block_entropy.IsAeonBeginning() && current.block_number != 0)
   {
-    ret = GetBlockPriorTo(ret, chain);
+    auto prior = GetBlockPriorTo(ret, chain);
+
+    if (!prior)
+    {
+      FETCH_LOG_ERROR(LOGGING_NAME,
+                      "Failed to find the beginning of the aeon when traversing the chain!");
+      throw std::runtime_error("Failed to traverse main chain");
```

Comment:
Are these handled correctly at the block coordinator level

Reply:
the BC catches all state exceptions, so it will not std::terminate if this were to throw. However, I believe it can only throw if you either try to update or verify a loose block from the BC, or remove a block from the chain elsewhere, which does not happen. So, I am  sure it will not throw this in the BC.

Reply:
Okay